sdk/inc/azure/core/az_credentials.h

```diff
@@ -104,27 +104,54 @@ typedef struct
     az_span client_id;
     az_span client_secret;
     az_span scopes;
+    az_span authority;
   } _internal;
 } az_credential_client_secret;
 
 /**
- * @brief az_credential_client_secret_init initializes an az_credential_client_secret instance
- * with the specified tenant ID, client ID and client secret.
+ * @brief Initializes an az_credential_client_secret instance with the specified tenant ID, client
+ * ID and client secret.
+ *
+ * @param out_credential Reference to an #az_credential_client_secret instance to initialize,
+ * @param tenant_id An Azure tenant ID.
+ * @param client_id An Azure client ID.
+ * @param client_secret An Azure client secret.
  *
- * @param out_credential reference to a az_credential_client_secret instance to initialize
- * @param tenant_id an Azure tenant ID
- * @param client_id an Azure client ID
- * @param client_secret an Azure client secret
  * @return An #az_result value indicating the result of the operation:
- *         - #AZ_OK if successful
- *         - Other error code if initialization failed
+ *         - #AZ_OK if successful.
+ *         - Other error code if initialization failed.
  */
 AZ_NODISCARD az_result az_credential_client_secret_init(
     az_credential_client_secret* out_credential,
     az_span tenant_id,
     az_span client_id,
     az_span client_secret);
 
+/**
+ * @brief Initializes an az_credential_client_secret instance with the specified tenant ID, client
+ * ID, client secret, and Azure AD authority URL.
+ *
+ * @param out_credential Reference to an #az_credential_client_secret instance to initialize,
+ * @param tenant_id An Azure tenant ID.
+ * @param client_id An Azure client ID.
+ * @param client_secret An Azure client secret.
+ * @param authority Authentication authority URL to set.
```

Comment:
To confirm, is that what this is called and referred to as (`authentication authority URL`)? Search results don't seem to refer to it that way:
https://www.google.com/search?q=Authentication+authority+URL&rlz=1C1GCEU_enUS820US821&oq=Authentication+authority+URL&aqs=chrome..69i57&sourceid=chrome&ie=UTF-8

Maybe it should be something like the following (granted, I am not the domain expert):
- Authority URL to set, used for authentication.
-  Authenticated authority URL to set.